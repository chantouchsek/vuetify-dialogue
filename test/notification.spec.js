import Vue from 'vue'
import Vuetify from 'vuetify/lib'
import { sleep } from './utils'
import VuetifyDialogPlugin from '../src'
Vue.use(Vuetify)
const vuetify = new Vuetify({})

describe('manager', () => {
  let manager

  beforeAll(() => {
    Vue.use(VuetifyDialogPlugin, { context: { vuetify } })
    manager = Vue.prototype.$dialog
  })

  test('Test notification component', async () => {
    const dlg = await manager.notification({
      text: 'Test notification',
      timeout: 100,
      waitForResult: false
    })
    await sleep(1)
    expect(dlg.vm.$el).toMatchSnapshot()
    await sleep(200)
    await Vue.nextTick()
    expect(document.body.innerHTML).toBe('')
  })

  test('Test success notification', async () => {
    const dlg = await manager.notify.success('Test notification', {
      timeout: 100,
      waitForResult: false
    })
    await sleep(1)
    expect(dlg.vm.$el).toMatchSnapshot()
    await sleep(200)
    await Vue.nextTick()
    expect(document.body.innerHTML).toBe('')
  })

  test('Test info notification', async () => {
    const dlg = await manager.notify.info('Test notification', {
      timeout: 1,
      waitForResult: false
    })
    await sleep(1)
    expect(dlg.vm.$el).toMatchSnapshot()
    await sleep(200)
    await Vue.nextTick()
    expect(document.body.innerHTML).toBe('')
  })

  test('Test error notification with top-left position', async () => {
    const dlg = await manager.notify.error('Test notification', {
      timeout: 100,
      position: 'top-left',
      waitForResult: false
    })
    await sleep(1)
    expect(dlg.vm.$el).toMatchSnapshot()
    await sleep(500)
    await Vue.nextTick()
    expect(document.body.innerHTML).toBe('')
  })

  test('Test warning notification with bottom-right position', async () => {
    const dlg = await manager.notify.warning('Test notification', {
      timeout: 100,
      position: 'bottom-right',
      waitForResult: false
    })
    await sleep(50)
    expect(dlg.vm.$el).toMatchSnapshot()
    await sleep(1000)
    await Vue.nextTick()
    expect(document.body.innerHTML).toBe('')
  })
})
