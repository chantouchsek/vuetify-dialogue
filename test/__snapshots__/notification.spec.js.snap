// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`manager Test error notification with top-left position 1`] = `
<div
  class="vuedl-notification left vuedl-notification-fade-enter vuedl-notification-fade-enter-active"
  role="alert"
  style="max-width: 500px; z-index: 2000; top: 10px;"
>
  <div
    class="v-alert v-sheet theme--dark error"
    role="alert"
    style="margin: 0px; min-width: 300px;"
  >
    <div
      class="v-alert__wrapper"
    >
      <i
        aria-hidden="true"
        class="v-icon notranslate v-alert__icon mdi mdi-alert theme--dark"
      />
      <div
        class="v-alert__content"
      >
        <div
          class="d-flex align-center"
        >
          <div
            class="mr-2"
          >
            
      Test notification
    
          </div>
           
          <!---->
        </div>
      </div>
      <button
        aria-label="Close"
        class="v-alert__dismissible v-btn v-btn--flat v-btn--icon v-btn--round theme--dark v-size--small"
        type="button"
      >
        <span
          class="v-btn__content"
        >
          <i
            aria-hidden="true"
            class="v-icon notranslate mdi mdi-close-circle theme--dark"
          />
        </span>
      </button>
    </div>
  </div>
   
  <!---->
</div>
`;

exports[`manager Test info notification 1`] = `
<div
  class="vuedl-notification right vuedl-notification-fade-leave vuedl-notification-fade-leave-active"
  role="alert"
  style="max-width: 500px; z-index: 2000; top: 10px;"
>
  <div
    class="v-alert v-sheet theme--dark info"
    role="alert"
    style="margin: 0px; min-width: 300px;"
  >
    <div
      class="v-alert__wrapper"
    >
      <i
        aria-hidden="true"
        class="v-icon notranslate v-alert__icon mdi mdi-information theme--dark"
      />
      <div
        class="v-alert__content"
      >
        <div
          class="d-flex align-center"
        >
          <div
            class="mr-2"
          >
            
      Test notification
    
          </div>
           
          <!---->
        </div>
      </div>
      <button
        aria-label="Close"
        class="v-alert__dismissible v-btn v-btn--flat v-btn--icon v-btn--round theme--dark v-size--small"
        type="button"
      >
        <span
          class="v-btn__content"
        >
          <i
            aria-hidden="true"
            class="v-icon notranslate mdi mdi-close-circle theme--dark"
          />
        </span>
      </button>
    </div>
  </div>
   
  <!---->
</div>
`;

exports[`manager Test notification component 1`] = `
<div
  class="vuedl-notification right vuedl-notification-fade-enter vuedl-notification-fade-enter-active"
  role="alert"
  style="max-width: 500px; z-index: 2000; top: 10px;"
>
  <div
    class="v-alert v-sheet theme--dark info"
    role="alert"
    style="margin: 0px; min-width: 300px;"
  >
    <div
      class="v-alert__wrapper"
    >
      <i
        aria-hidden="true"
        class="v-icon notranslate v-alert__icon mdi mdi-information theme--dark"
      />
      <div
        class="v-alert__content"
      >
        <div
          class="d-flex align-center"
        >
          <div
            class="mr-2"
          >
            
      Test notification
    
          </div>
           
          <!---->
        </div>
      </div>
      <button
        aria-label="Close"
        class="v-alert__dismissible v-btn v-btn--flat v-btn--icon v-btn--round theme--dark v-size--small"
        type="button"
      >
        <span
          class="v-btn__content"
        >
          <i
            aria-hidden="true"
            class="v-icon notranslate mdi mdi-close-circle theme--dark"
          />
        </span>
      </button>
    </div>
  </div>
   
  <!---->
</div>
`;

exports[`manager Test success notification 1`] = `
<div
  class="vuedl-notification right vuedl-notification-fade-enter vuedl-notification-fade-enter-active"
  role="alert"
  style="max-width: 500px; z-index: 2000; top: 10px;"
>
  <div
    class="v-alert v-sheet theme--dark success"
    role="alert"
    style="margin: 0px; min-width: 300px;"
  >
    <div
      class="v-alert__wrapper"
    >
      <i
        aria-hidden="true"
        class="v-icon notranslate v-alert__icon mdi mdi-check-circle theme--dark"
      />
      <div
        class="v-alert__content"
      >
        <div
          class="d-flex align-center"
        >
          <div
            class="mr-2"
          >
            
      Test notification
    
          </div>
           
          <!---->
        </div>
      </div>
      <button
        aria-label="Close"
        class="v-alert__dismissible v-btn v-btn--flat v-btn--icon v-btn--round theme--dark v-size--small"
        type="button"
      >
        <span
          class="v-btn__content"
        >
          <i
            aria-hidden="true"
            class="v-icon notranslate mdi mdi-close-circle theme--dark"
          />
        </span>
      </button>
    </div>
  </div>
   
  <!---->
</div>
`;

exports[`manager Test warning notification with bottom-right position 1`] = `
<div
  class="vuedl-notification right"
  role="alert"
  style="max-width: 500px; z-index: 2000; bottom: 10px;"
>
  <div
    class="v-alert v-sheet theme--dark warning"
    role="alert"
    style="margin: 0px; min-width: 300px;"
  >
    <div
      class="v-alert__wrapper"
    >
      <i
        aria-hidden="true"
        class="v-icon notranslate v-alert__icon mdi mdi-exclamation theme--dark"
      />
      <div
        class="v-alert__content"
      >
        <div
          class="d-flex align-center"
        >
          <div
            class="mr-2"
          >
            
      Test notification
    
          </div>
           
          <!---->
        </div>
      </div>
      <button
        aria-label="Close"
        class="v-alert__dismissible v-btn v-btn--flat v-btn--icon v-btn--round theme--dark v-size--small"
        type="button"
      >
        <span
          class="v-btn__content"
        >
          <i
            aria-hidden="true"
            class="v-icon notranslate mdi mdi-close-circle theme--dark"
          />
        </span>
      </button>
    </div>
  </div>
   
  <!---->
</div>
`;
